### PR TITLE
Editorial: refactor common behaviour of SetFunctionName

### PR DESCRIPTION
**Pull request by devsnek**

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -7504,9 +7504,10 @@ <h1>MakeMethod ( _F_, _homeObject_ )</h1>
       <h1>SetFunctionName ( _F_, _name_ [ , _prefix_ ] )</h1>
       <p>The abstract operation SetFunctionName requires a Function argument _F_, a String or Symbol argument _name_ and optionally a String argument _prefix_. This operation adds a `name` property to _F_ by performing the following steps:</p>
       <emu-alg>
-        1. Assert: _F_ is an extensible object that does not have a `name` own property.
+        1. Assert: _F_ is an extensible object.
         1. Assert: Type(_name_) is either Symbol or String.
         1. Assert: If _prefix_ is present, then Type(_prefix_) is String.
+        1. If ! HasOwnProperty(_F_, `"name"`) is *true*, return.
```
Comment (ljharb) [inline spec.html]: I’m not sure it’s as clear to have an operation named “set function name” not always set the function name, although i see how this DRYs up the steps. What do you think?

Comment (jmdyck) [reply spec.html]: The status quo has '?' before the corresponding calls to `HasOwnProperty`. I don't think you're justified in changing that to '!' here.

Comment (zenparsing) [reply spec.html]: I'm wondering the same thing as @ljharb. I like the general idea of cleaning things up with refactoring, but for specs we should prioritize clarity.

Comment (jmdyck) [inline spec.html]: Where the replaced code has a call to `HasOwnProperty`, that call is always preceded by '?'. I don't think you're justified in changing that to '!' here. Moreover, in those cases, the resulting call to `SetFunctionName` should be preceded by '?'.

**Comment by TimothyGu** [comment]
There exists usage of the SetFunctionName abstract operation in other non-algorithmic parts of the spec, such as in the [Web IDL specification](https://heycam.github.io/webidl/). If this pull request were to land, modifications to that specification to use the new EnsureFunctionName would be nice.

Alternatively, we could provide a compatibility SetFunctionName abstract operation in ECMA-262 still.

**Comment by devsnek** [comment]
I think 1372 does what I was looking to do with this. Any reason anyone sees to keep this open?

**Pull request closed without merging**